[Server] Fix: WMS GetStyles contains previous SLD_BODY requests

rldhont opened the pull request:
## Description

The WMS Server restorer does not remove styles created from SLD provided in WMS requests.

The custom property that contains the SLD style name has to be checked before the layer style is restored to the project current style.

Fundex by Ifremer

pblottiere commented on src/server/services/wms/qgswmsrenderer.cpp:
```diff
@@ -153,6 +153,9 @@ namespace QgsWms
 
+    // restore layers
+    restorer.reset( new QgsWmsRestorer( mContext ) );
+    // return
```
I'm not sure to understand what you're trying to achieve. The `restorer` is a unique pointer so the destructor is called once out of scope. And it's actually in the destructor that layers are restored to their original state.

So unless I'm missing something, I'd said that it's not useful.

rldhont replied:
@pblottiere If you take alook at the test I have added, without an explicite reset:
1. GetStyles contains only QGIS styles
2. GetMap with SLD_BODY
3. GetStyles contains QGIS styles and the UserStyle of the GetMap with SLD_BODY

rldhont replied:
@pblottiere I have found the issue and fix it.

rldhont committed:
[Server] Fix: WMS GetStyles contains previous SLD_BODY requests

The WMS Server restorer does not remove styles created from SLD provided in WMS requests.

The custom property that contains the SLD style name has to be checked before the layer style is restored to the project current style.

Fundex by Ifremer

nyalldawson commented:
@pblottiere is this for merging for 3.22?

pblottiere commented on src/server/services/wms/qgswmsrestorer.cpp:
```diff
@@ -74,18 +74,19 @@ QgsLayerRestorer::~QgsLayerRestorer()
+    // Then restore the previous style
+    const QgsLayerSettings settings = mLayerSettings[layer];
+    layer->styleManager()->setCurrentStyle( settings.mNamedStyle );
+    layer->setName( mLayerSettings[layer].name );
```
Do you know why the order is important?

rldhont replied:
Yes, the custom variables are stored in the style, so the style has to be reset after getting the variables.

pblottiere replied:
ok, thanks for the explanation :+1:

rldhont commented:
> @pblottiere is this for merging for 3.22?

@nyalldawson I hope so, it is a bugfix.
